extract_unique_sensors_01.py

```diff
@@ -86,9 +86,9 @@ def extract_all_sensors(sensors_data, sensor_name):
     assert isinstance(sensor_name, str), "Expected a string."
     assert isinstance(sensors_data, list), "Expected a list."
     candidate_lines = []
-    [
-        candidate_lines.append(candidate_line)for sensor_data in sensors_data
-        if sensor_data.startswith(sensor_name) for candidate_line in sensor_data.split()
+    candidate_lines = [
+        candidate_line for sensor_data in sensors_data if sensor_data.startswith(sensor_name)
```

Comment:
Hi Lebo

Nice work implementing list comps in your project.
I would advise you to change the name of your iter_var `sensor_data` to be something different from the sequence you are iterating through, just to avoid confusion and make it easier to read.
E.g: `candidate_line for sensor in sensors_data if sensor.startswith(sensor_name)`